common/path.cpp

```diff
@@ -178,8 +231,107 @@ Path Path::join(const String &str, char separator) const {
 
 Path Path::join(const char *str, char separator) const {
 	Path temp(*this);
-	temp.joinInPlace(str, DIR_SEPARATOR);
+	temp.joinInPlace(str, separator);
 	return temp;
 }
 
+Common::Array<Common::String> Path::splitComponents() const {
+	Common::Array<Common::String> res;
+	Common::String cur;
+	for (uint i = 0; i < _str.size(); i++) {
+		if (_str[i] == ESCAPER) {
+			i++;
+			if (_str[i] == ESCAPE_SLASH)
+				cur += '/';
+			else if (_str[i] == ESCAPE_SEPARATOR) {
+				res.push_back(cur);
+				cur = "";
+			}
+		} else
+			cur += _str[i];
+	}
+
+	res.push_back(cur);
+
+	return res;
+}
+
+Path Path::punycodeDecode() const {
+	Common::Array<Common::String> c = splitComponents();
```

Comment:
StringArray here and everywhere else below

Reply:
Done